source/server/syscfg_library.cpp

```diff
@@ -142,4 +145,42 @@ NISysCfgStatus SysCfgLibrary::GetResourceProperty(
   return function_pointers_.GetResourceProperty(resource_handle, property_ID, value);
 }
 
+NISysCfgStatus SysCfgLibrary::GetInstalledSoftwareComponents(
+    NISysCfgSessionHandle session_handle,
+    NISysCfgIncludeComponentTypes item_types,
+    NISysCfgBool cached,
+    NISysCfgEnumSoftwareComponentHandle* component_enum_handle)
+{
+  if (!function_pointers_.GetInstalledSoftwareComponents) {
+    throw LibraryLoadException(kSysCfgApiNotInstalledMessage);
+  }
+  return function_pointers_.GetInstalledSoftwareComponents(session_handle, item_types, cached, component_enum_handle);
+}
+
+NISysCfgStatus SysCfgLibrary::ResetEnumeratorGetCount(
+    void* enum_handle,
+    unsigned int* count
+)
+{
+  if (!function_pointers_.ResetEnumeratorGetCount) {
+    throw LibraryLoadException(kSysCfgApiNotInstalledMessage);
+  }
+  return function_pointers_.ResetEnumeratorGetCount(enum_handle, count);
+}
+
+NISysCfgStatus SysCfgLibrary::NextComponentInfo(
+    NISysCfgEnumSoftwareComponentHandle component_enum_handle,
+    char* id,
+    char* version,
+    char* title,
+    NISysCfgComponentType* itemType,
+    char** detailedDescription
+  )
+  {
+    if (!function_pointers_.NextComponentInfo) {
+    throw LibraryLoadException(kSysCfgApiNotInstalledMessage);
+  }
```

Comment:
Formatting is off here. Do you have the [clang auto-formatter set up](https://github.com/ni/grpc-device/wiki/Development-Tips-and-Tricks#clang-format-on-save)?